File: frontend/src/types/review.ts
import type { PropertyHost } from "./property";

export type Review = {
  ID: number;
  CreatedAt: string;
  property_id: number;
  booking_id: number;
  user_id: number;
  user: PropertyHost;
  rating: number;
  comment: string;
  cleanliness: number;
  accuracy: number;
  communication: number;
  location: number;
  check_in: number;
  value: number;
};

export type ReviewInput = {
  booking_id: number;
  rating: number;
  comment?: string;
  cleanliness?: number;
  accuracy?: number;
  communication?: number;
  location?: number;
  check_in?: number;
  value?: number;
};

File: backend/internal/model/review.go
package model

import "gorm.io/gorm"

type Review struct {
	gorm.Model

	PropertyID uint     `gorm:"index;not null" json:"property_id"`
	Property   Property `gorm:"foreignKey:PropertyID" json:"-"`

	BookingID uint    `gorm:"uniqueIndex;not null" json:"booking_id"`
	Booking   Booking `gorm:"foreignKey:BookingID" json:"-"`

	UserID uint `gorm:"index;not null" json:"user_id"`
	User   User `gorm:"foreignKey:UserID" json:"user"`

	Rating  int    `gorm:"not null" json:"rating"`
	Comment string `gorm:"type:text" json:"comment"`

	Cleanliness   int `json:"cleanliness"`
	Accuracy      int `json:"accuracy"`
	Communication int `json:"communication"`
	Location      int `json:"location"`
	CheckIn       int `json:"check_in"`
	Value         int `json:"value"`
}
